build.yaml

```diff
@@ -169,6 +162,105 @@ common_cli:
       fi
     }
 
+    # Retries a command if it fails, using the exponential backoff algorithm.
+    # Usage:
+    # retry_n_times_with_timeout <max retries> <timeout in seconds> command
+    #
+    function retry_n_times_with_timeout {
+      set +e
+      max_tries=\${1}
+      seconds=\${2}
+      shift 2
+
+      let delay=15
+      let tries=0
+      while [ \${tries} -lt \${max_tries} ]; do
+        echo "executing...  \${*}"
+        if (timeout --foreground --signal=SIGKILL \${seconds} bash -c "\${*}"); then
+            break
+        fi
+        echo Command failed, waiting \${delay}...
+        sleep \${delay}
+        let delay=delay*2
+        let tries=tries+1
+
+        # abort on too many failures
+        if [ \${tries} -eq \${max_tries} ]; then
+          echo "too many failed retries... aborting"
+          exit 1
+        fi
+
+      done
+    }
+
+    # uploads a file to S3, retrying on failure.
+    # usage:
+    # s3_upload <bucket> <filename>
+    function s3_upload() {
+      echo "uploading \${2} to S3 bucket ${1} ..."
+      retry_n_times_with_timeout 3 900 \
+        aws --region us-west-2 s3 cp "\${2}" s3://\${1}/
```

Comment:
Why are we using this specific region? 

Could we please make this a constant that gets referred to, with some documentation on region choice (even if it's just "arbitrary region to use for s3 uploads")?